app/controllers/posts_controller.rb

```diff
@@ -6,9 +6,28 @@ class PostsController < ApplicationController
 
   def index
     if current_user.admin?
-      @posts = Post.pending
+      if params[:type] == "user"
+        if params[:most]
+          post_id = React.group(:post_id).order("count_all DESC").count.first[0]
+          post = Post.where(id: post_id).first
+          @res = User.where(id: post.user_id)
+        else
+          @pagy, @res = pagy_countless(User.all, items:10)
+        end
+      else
+        case params[:status]
+        when "pending"
+          @pagy, @res = pagy_countless(Post.pending, items:10)
+        when "published"
+          @pagy, @res = pagy_countless(Post.published, items:10)
+        when "draft"
+          @pagy, @res = pagy_countless(Post.draft, items:10)
+        else
+          @pagy, @res = pagy_countless(Post.all, items:10)
+        end
```

Comment:
We can use metaprogramming here to avoid repeating
```
if Post.statuses.keys.include?(params[:status])
  @pagy, @res = pagy_countless(Post.send(params[:status]), items:10)
else
  @pagy, @res = pagy_countless(Post.all, items:10)
end
```